<?xml version="1.0" ?>

<!-- Not shown, but you can also include test bags to get a testing error!-->
<launch>
	
  <arg name="save_file" value="$(find leg_tracker)/config/trained_leg_detector_RForest_sim.yaml" />

  <arg name="b_bag_1" value="$(find leg_tracker)/rosbags/sim_data/train-back-scaled-simulated-2023-05-08-17-21-31.bag" />
  <arg name="f_bag_1" value="$(find leg_tracker)/rosbags/sim_data/train-front-scaled-simulated-2023-05-08-17-21-31.bag" />
  <arg name="b_bag_2" value="$(find leg_tracker)/rosbags/sim_data/train-back-scaled-simulated-2023-05-14-22-40-36.bag" />
  <arg name="f_bag_2" value="$(find leg_tracker)/rosbags/sim_data/train-front-scaled-simulated-2023-05-14-22-40-36.bag" />

  <arg name="b_bag_3" value="$(find leg_tracker)/rosbags/sim_data/train-back-scaled-simulated-2023-03-21-16-24-29.bag" />
  <arg name="f_bag_3" value="$(find leg_tracker)/rosbags/sim_data/train-front-scaled-simulated-2023-03-21-16-24-29.bag" />
  <arg name="b_bag_4" value="$(find leg_tracker)/rosbags/sim_data/train-back-scaled-simulated-2023-03-21-16-26-30.bag" />
  <arg name="f_bag_4" value="$(find leg_tracker)/rosbags/sim_data/train-front-scaled-simulated-2023-03-21-16-26-30.bag" />
  <arg name="b_bag_5" value="$(find leg_tracker)/rosbags/sim_data/2023-05-28-18-38-08.bag" />
  <arg name="b_bag_6" value="$(find leg_tracker)/rosbags/sim_data/2023-05-28-18-32-42.bag" />


  <arg name="pos_scan_topic" value="/training_scan" />
  <arg name="neg_scan_topic" value="/neg_scan" />
  <arg name="front_topic" value="/f_scan"/>
  <arg name="back_topic" value="/b_scan"/>

  <arg name="test_bag_1" value="$(find leg_tracker)/rosbags/sim_data/test-back-scaled-simulated-2023-05-08-17-19-34.bag" />
  <arg name="test_bag_2" value="$(find leg_tracker)/rosbags/sim_data/test-front-scaled-simulated-2023-05-08-17-19-34.bag" />

  <node pkg="leg_tracker" type="train_leg_detector" name="train_leg_detector" required="true"
      args="--pos $(arg b_bag_1) $(arg pos_scan_topic) 
            --pos $(arg f_bag_1) $(arg pos_scan_topic)
            --pos $(arg b_bag_2) $(arg pos_scan_topic) 
            --pos $(arg f_bag_2) $(arg pos_scan_topic) 
            --pos $(arg b_bag_3) $(arg pos_scan_topic) 
            --pos $(arg f_bag_3) $(arg pos_scan_topic) 
            --pos $(arg b_bag_4) $(arg pos_scan_topic) 
            --pos $(arg f_bag_4) $(arg pos_scan_topic)
            --neg $(arg b_bag_1) $(arg neg_scan_topic) 
            --neg $(arg f_bag_1) $(arg neg_scan_topic)
            --neg $(arg b_bag_2) $(arg neg_scan_topic) 
            --neg $(arg f_bag_2) $(arg neg_scan_topic) 
            --neg $(arg b_bag_3) $(arg neg_scan_topic) 
            --neg $(arg f_bag_3) $(arg neg_scan_topic) 
            --neg $(arg b_bag_4) $(arg neg_scan_topic) 
            --neg $(arg f_bag_4) $(arg neg_scan_topic)
            --neg $(arg b_bag_5) $(arg front_topic) 
            --neg $(arg b_bag_5) $(arg back_topic)
            --neg $(arg b_bag_6) $(arg front_topic) 
            --neg $(arg b_bag_6) $(arg back_topic)
            --test_neg $(arg test_bag_1) $(arg neg_scan_topic)   
            --test_neg $(arg test_bag_2) $(arg neg_scan_topic)              
            --test_pos $(arg test_bag_1) $(arg pos_scan_topic)   
            --test_pos $(arg test_bag_2) $(arg pos_scan_topic)
            --save_file $(arg save_file)" 
            output="screen">
      <param name="min_points_per_cluster" value="2"/>

  </node> 
</launch>


